### PR TITLE
React Calculator Project - Milestone 3 finished

### PR DESCRIPTION
**Pull request by Flakster**
## This pull request includes

* Review of the requirements for the milestone 3

### CODE REVIEW

**Review by SpaYco — CHANGES_REQUESTED** [review]
# Changes Requested 

Hello Team! 👋 
Awesome job so far!! 👍 you still need to make some changes though! 

## Required Changes ♻️ 

* [ ] The styled version of the calculator should look like this: [Calculator](https://github.com/microverseinc/project-react-calculator/blob/master/images/calculator.png) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone3/calculator-style.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/44896093/))
* [ ] `App`, `Display` and `ButtonPanel` components use flex display mode ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone3/useFlex.png))
* [ ] calculator width equals 700px ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone3/700Wide.png))

##### `Display` component

* [ ] has a gray background ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone3/display-bg.png))
* [ ] has a height of 100px ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone3/display-height.png))
* [ ] result is presented with white and bold text ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone3/display-text.png))
* [ ] has a padding from the edges ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone3/display-padding.png))
* [ ] result is aligned to the right ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone3/display-alignment.png))

##### `ButtonPanel` component

* [ ] each row's height is 100px ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone3/rowHeight.png))

##### `Button` component

* [ ] buttons take 25% of container width (except `0` button) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone3/25Wide.png))
* [ ] buttons have borders ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone3/ButtonsBoarder.png))
* [ ] buttons text is centered and black ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone3/TextBlack.png))
* [ ] it accepts `color` and `wide` as an incoming props ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone3/colorAndWideProps.png))
* [ ] default `color` for button is orange ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone3/colorProps.png))
* [ ] `wide` is boolean, true means that button take 50% of space (used for `0` button) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone3/WideProps.png))


### please make the appropriate changes and ask for another code review ✔️ 

Happy Coding!

Best, 
Aziz Mejri 
_contact me on [twitter](https://twitter.com/iSpaYco) or slack (@SpaYco) if you need any help_

**Comment by Flakster** [comment]
I think the Calculator should fulfill all the requirements that you mentioned but in Milestone 3 (intended to be completed in 5 minutes) requirements says:

Milestone 3
[Solo Project]

Your components are set and ready to work, but they don’t look really pretty, do they?
This milestone is here only to remind you that you should not do anything about styling for this project at this moment. Do not worry though! You will be asked to make it pretty very soon!

Please clarify what is required for this milestone

**Comment by SpaYco** [comment]
I replied on slack @Flakster :)

**Comment by Flakster** [comment]
## This new submission includes
* Styling of the components according to the given design
* Use of CSS Flexbox
* Pass of wide and color features as props
* Setting of default values for props

File: src/components/Button.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+Button.propTypes = {
+  buttonName: PropTypes.string,
+  size: PropTypes.string,
+  color: PropTypes.string,
+};
+Button.defaultProps = {
+  buttonName: '',
+  size: 'small',
+  color: 'red',
+};
```
Comment (SpaYco) [inline src/components/Button.js]: * [x] default `color` for button is orange ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone3/colorProps.png))

File: src/components/Button.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import PropTypes from 'prop-types';
+
+function Button(props) {
+  const { buttonName, size, color } = props;
+  const classes = `button-box ${color}-button ${size}-button`;
+  return (
+    <button type="button" className={classes}>
+      { buttonName }
+    </button>
+  );
```
Comment (SpaYco) [inline src/components/Button.js]: * [x] it accepts `wide` as an incoming props ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone3/colorAndWideProps.png))
* [x] `wide` is boolean, true means that button take 50% of space (used for `0` button) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone3/WideProps.png))

**Review by SpaYco — CHANGES_REQUESTED** [review]
# Changes Requested 

Hello! 👋 
Awesome job so far!! 👍 you still need to make some changes though! 

## Required Changes ♻️ 

### please make the appropriate changes and ask for another code review ✔️ 

Happy Coding!

Best, 
Aziz Mejri 
_contact me on [twitter](https://twitter.com/iSpaYco) or slack (@SpaYco) if you need any help_

**Comment by Flakster** [comment]
## This new submission includes
* Change props definition and default values according to the requiirements

**Review by teekaytech — APPROVED** [review]
### Status: Approved :green_circle: 

**Congratulations @Flakster :clap:, this milestone has passed all the mandatory requirements. It is therefore approved :heavy_check_mark:**

Good luck with your next milestone.

* * * 
**Taofeek Olalere
[Twitter](https://twitter.com/ola_lere) | [Github](https://github.com/teekaytech) | @taofeekolalere on Slack**
__Happy Coding :smiley:__

**Pull request merged**